Makefile

```diff
@@ -67,10 +68,28 @@ local: ## - Build local binary for local environment (bin/fleet-server)
 	go build $(if $(DEV),-tags="dev",) -gcflags="${GCFLAGS}" -ldflags="${LDFLAGS}" -o ./bin/fleet-server .
 	@printf "${CMD_COLOR_ON} Binaries in ./bin/\n${CMD_COLOR_OFF}"
 
+
+.PHONY: cover-e2e-binaries
+cover-e2e-binaries: ## - Build binaries for the test-e2e target with the go 1.20+ cover flag
+	SNAPSHOT=true $(MAKE) cover-$(shell go env GOOS)/$(shell go env GOARCH)
+	SNAPSHOT=true $(MAKE) cover-linux/$(shell go env GOARCH)
+
+.PHONY: $(COVER_TARGETS)
+$(COVER_TARGETS): cover-%: ## - Build a binary with the -cover flag for integration testing
+	@mkdir -p build/cover
+	$(eval $@_OS := $(firstword $(subst /, ,$(lastword $(subst cover-, ,$@)))))
+	$(eval $@_GO_ARCH := $(lastword $(subst /, ,$(lastword $(subst cover-, ,$@)))))
+	$(eval $@_ARCH := $(TARGET_ARCH_$($@_GO_ARCH)))
+	$(eval $@_BUILDMODE:= $(BUILDMODE_$($@_OS)_$($@_GO_ARCH)))
+	GOOS=$($@_OS) GOARCH=$($@_GO_ARCH) go build $(if $(DEV),-tags="dev",) -cover -coverpkg=./... -gcflags="${GCFLAGS}" -ldflags="${LDFLAGS}" $($@_BUILDMODE) -o build/cover/fleet-server-$(VERSION)-$($@_OS)-$($@_ARCH)/fleet-server .
+ifeq ($($@_OS),windows) # FIXME this isn't working on windows builds?
```

Comment:
My knowledge of make is a bit weak, but the rename here wasn't working for me. Does anyone know the syntax I can use here?